lib/datapacktypes/flexcard.js

```diff
@@ -0,0 +1,287 @@
+
+            const flexCardsCompiler = await VlocityUtils.loadCompiler(compilerName,
+                                            jobInfo, packageVersion, namespace);
+            if(flexCardsCompiler){
+                for (let i = 0; i < idsArray.length; i++) { 
+                    let flexCardID = idsArray[i];
+                    await this.compileFlexCardsLwc(flexCardID,jobInfo,flexCardsCompiler) ;
+                } 
+            } else{
+                hasFlexCardCompiler= false;
+            }
+
+        } catch (e) {
+            hasFlexCardCompiler= false;
+            VlocityUtils.error('Error while loading Flexcard Compiler', e);
+        }
+
+         // If we were unable to load the compiler package, use the legacy method
+        if(!hasFlexCardCompiler){
+            await this.flexCardDeployWithPuppeteer(jobInfo) ;
+            return;
+        }
+    }
+
+    if(!jobInfo.haveCompiledUILayouts) {
+        let layoutOb = new VlocityUILayout(this.vlocity);
+        await layoutOb.onDeployFinish(jobInfo);
+    }
+} 
+FlexCard.prototype.compileFlexCardsLwc = async function (flexCardID,jobInfo,flexCardsCompiler) {
+    VlocityUtils.report('Compiling Flexcard with  FlexCard ID: ',flexCardID); 
+    try { 
+        let tempSfdxFolder = path.join(jobInfo.tempFolder, 'tempDeployLWC', 'salesforce_sfdx/lwc');
+          
+        const fCCompiler = new flexCardsCompiler(this.vlocity.namespace, this.vlocity.jsForceConnection,
+                                                 this.vlocity.jsForceConnection.instanceUrl, false);  
+        await fCCompiler.compileMetadataWithId(flexCardID, tempSfdxFolder);
+        jobInfo.deployGeneratedLwc = true; 
+    }
+    catch (e) {
+        hasFlexCardCompiler= false;
+        VlocityUtils.error(e);
+    }
+    
+} 
+
+FlexCard.prototype.afterActivationSuccess = async function(inputMap) {
+    var jobInfo = inputMap.jobInfo;   
+    var dataPack = inputMap.dataPack;
+    var data = dataPack.VlocityDataPackData['%vlocity_namespace%__VlocityCard__c'] || dataPack.VlocityDataPackData['OmniUiCard'];
+    var cardSourceKey = data[0].VlocityRecordSourceKey;
+    var definition = data[0]['%vlocity_namespace%__Definition__c'] || data[0]['PropertySetConfig'];
+    var isFlexCard = definition.includes('"isFlex":true,');
+    var isLWC = definition.includes('"enableLwc":true,');
+    if (isFlexCard) {
+        if (!jobInfo.flexCardsToCompile) {
+            jobInfo.flexCardsToCompile = {};
+        }
+        VlocityUtils.log('LWC for FlexCard will be compiled at the end', dataPack.VlocityDataPackKey);
+        jobInfo.flexCardsToCompile[jobInfo.sourceKeyToRecordId[cardSourceKey]] = dataPack.VlocityDataPackKey;
+    } else if (isLWC && jobInfo.lookForParentLayoutsForlWCCards) {
+        var parentUILayouts = await this.addParentUILayoutForCard(jobInfo,dataPack.VlocityDataPackKey);
+    }
+}
+
+
+FlexCard.prototype.addParentUILayoutForCard = async function(jobInfo,carKey) {
```

Comment:
This parent layout stuff should be removed.

Reply:
i removed the method and call , could you please review